core/controllers/acl_decorators.py

```diff
@@ -2503,21 +2519,25 @@ def test_can_edit_entity(self, entity_type, entity_id, **kwargs):
             PageNotFoundException: The given page cannot be found.
         """
         arg_swapped_handler = lambda x, y, z: handler(y, x, z)
-        if entity_type == feconf.ENTITY_TYPE_EXPLORATION:
-            # This swaps the first two arguments (self and entity_type), so
-            # that functools.partial can then be applied to the leftmost one to
-            # create a modified handler function that has the correct signature
-            # for can_edit_question().
-            reduced_handler = functools.partial(
-                arg_swapped_handler, feconf.ENTITY_TYPE_EXPLORATION)
+        # This swaps the first two arguments (self and entity_type), so
+        # that functools.partial can then be applied to the leftmost one to
+        # create a modified handler function that has the correct signature
+        # for the corresponding decorators.
+        reduced_handler = functools.partial(
+            arg_swapped_handler, entity_type)
+        if (entity_type == feconf.ENTITY_TYPE_EXPLORATION or
+                entity_type == fs_domain.ENTITY_TYPE_EXPLORATION):
```

Comment:
Why are there 2 constants corresponding to ENTITY_TYPE_EXPLORATION?

Reply:
The one in feconf is used for distinguising between exploration and question in learner answer details submission, while the one in fs_domain is used for distinguishing between entities when uploading images.

Reply:
Actually, this is weird. Can we unify them please? Let's just have one set of constants in feconf.py.

Reply:
Done